documentation/rpi3.md

```diff
@@ -0,0 +1,144 @@
+also know that there is a strong interest among developers, students,
+researches to start work and learn more about TEE's using a Raspberry Pi. So
+instead of delaying this, we have decided to make what we have available right
+away. Hopefully there will be some enthusiast that will help out making proper
+upstream patches sooner or later.
+
+| Project | Base fork | What to do |
+|--------|--------|--------|
+| build | the official build master branch | Rebase and do a pull request |
+| optee_os | the official optee_os master branch | Rebase and do a pull request |
+| linux | https://github.com/Electron752/linux.git commit: 51d1fa5c3208f15e80d25d85ce03330909916cc8 | Two things here. 1. The base is a fork itself and should be upstreamed. 2. The current OP-TEE patches should be replaced with cherry-picked patches from the official OP-TEE Linux kernel branch |
+| arm-trusted-firmware | https://github.com/96boards-hikey/arm-trusted-firmware commit: bdec62eeb8f3153a4647770e08aafd56a0bcd42b | This should instead be based on the official OP-TEE fork or even better the official ARM repository. The patch itself should also be upstreamed. |
+| U-boot | https://github.com:linaro-swg/u-boot.git | This is just a mirror of the official U-boot git. The patches should be upstreamed. |
+| OpenOCD | TBD | TBD |
+
+# 3. Build instructions
+- First thing to pay attention to are the prerequisites stated
+  [here](https://github.com/OP-TEE/optee_os#41-prerequisites) in the README.md
+  file. If you forget that, then you can get all sorts of strange errors.
+
+- From the [README.md](https://github.com/OP-TEE/optee_os#5-repo-manifests),
+  you should follow section 5.1, 5.2. In short if you have repo installed, what
+  you need to do is something like this:
+```
+$ mkdir rpi3
+$ cd rpi3
+$ repo init -u https://github.com/OP-TEE/manifest.git -m rpi3_experimental.xml
+$ repo sync -j3
+```
+  Now it's probably a good idea to read the [Tips and tricks](https://github.com/OP-TEE/optee_os#58-tips-and-tricks)
+  section also, since that will save a lot of time in the long run.
+
+- Next step is to get the toolchains
+```
+$ cd build
+$ make toolchains
+```
+
+- Then it is time to build everything. Note that the initial build will download
+  a couple of files, like the official Raspberry Pi 3 firmware, the overlay root
+  fs etc. However, that is only done once, so subsequent builds won't re-download
+  them again (as long as you don't delete them).
+```
+$ make all -j4
+```
+
+- The last step is to partition and format the memory card and to put the files
+  onto the same. That is something we don't want to automate, since if anything
+  goes wrong, in worst case it might wipe one of your regular hard disks. Instead
+  what we have done, is that we have created another makefile target that will tell
+  you exactly what to do. Run that command and follow the instructions there.
+```
+$ make img-help
```

Comment:
Wouldn't documentation be more meaningful in the documentation rather than as a makefile target?